projects/siren/index.js

```diff
@@ -3,61 +3,115 @@
+    ethereumBlock,
+  });
 
-  async function tvl(timestamp, block) {
-    const {
-      amms,
-    } = await request(GRAPH_URL, GET_POOLS, {
-      block,
-    });
+  const mainnet_result = {
+    '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': '0'
+  };
 
-    const result = {
-      '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48': '0'
-    };
+      for (let i = 0; i < mainnet_amms['amms'].length; i++) {
+
+      const amm = mainnet_amms['amms'][i];
 
-    for (let i = 0; i < amms.length; i++) {
-      const amm = amms[i];
       const collateralToken = amm['collateralToken']['id'];
 
       // Get collateral in AMM
       const response = await sdk.api.erc20.balanceOf({
-        block,
+        block: ethereumBlock,
         target: collateralToken,
         owner: amm['id'],
+        chain: "ethereum"
       });
 
-      if (!result[collateralToken]) {
-        result[collateralToken] = '0';
+      if (!mainnet_result[collateralToken]) {
+        mainnet_result[collateralToken] = '0';
       }
-      result[collateralToken] = BigNumber(result[collateralToken]).plus(response.output).toFixed();
+      mainnet_result[collateralToken] = BigNumber(mainnet_result[collateralToken]).plus(response.output).toFixed();
 
       // Get collateral in Markets
       for (let im = 0; im < amm.markets.length; im++) {
         const market = amm.markets[im];
 
         const response = await sdk.api.erc20.balanceOf({
-          block,
+          block: ethereumBlock,
           target: collateralToken,
           owner: market['id'],
+          chain: "ethereum"
         });
 
-        result[collateralToken] = BigNumber(result[collateralToken]).plus(response.output).toFixed();
+        mainnet_result[collateralToken] = BigNumber(mainnet_result[collateralToken]).plus(response.output).toFixed();
       }
     }
+    return mainnet_result;
+
+}
+
+async function calculatePolygonTVL(timestamp, block, chainBlocks) {
+
+  const polygonBlock = chainBlocks['polygon'];
+
+  const polygon_amms = await request(POLYGON_GRAPH_URL, GET_POOLS_POLYGON, {
+    polygonBlock,
+  });
+
+  const polygon_result = {
+    'polygon:0x2791bca1f2de4661ed88a30c99a7a9449aa84174': '0'
+  };
+
+  for (let i = 0; i < polygon_amms['amms'].length; i++) {
+    const amm = polygon_amms['amms'][i];
+    const collateralToken = amm['collateralToken']['id'];
+
+    // Get collateral in AMM
+    const response = await sdk.api.erc20.balanceOf({
+        block: polygonBlock,
+        target: collateralToken,
+        owner: amm['id'],
+        chain: "polygon"
+      });
+
+    if (!polygon_result['polygon:'+collateralToken]) {
+      polygon_result['polygon:'+collateralToken] = '0';
+    }
+    polygon_result['polygon:'+collateralToken] = BigNumber(polygon_result['polygon:'+collateralToken]).plus(response.output).toFixed();
+
+    // Get collateral in Series
+    for (let im = 0; im < amm.series.length; im++) {
+
+      const response = await sdk.api.erc20.balanceOf({
```

Comment:
this is going to get the collateral token balance on the SeriesController, but recall we don't store any tokens on the SeriesController. I think what you want here is the SeriesVault, which is what stores the collateral locked in the protocol used to cover the options.